SEGFAULT in internal JVM when running scalafmt in large filesets

**Describe the bug**
When running `./pants fmt ::` on a codebase with a large set of Scala files (more than 1000), the internal JVM used by Pants (AdoptJDK 11) occasionally segfaults with the following error message:

```
12:19:06.67 [ERROR] Completed: Lint with scalafmt - scalafmt failed (exit code -6).
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f8473fed6f1, pid=6524, tid=6567
#
# JRE version: OpenJDK Runtime Environment (11.0+28) (build 11+28)
# Java VM: OpenJDK 64-Bit Server VM (11+28, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0xbf46f1]  Node::add_req(Node*)+0xc1
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P %E" (or dumping to /tmp/process-executionzWXy23/core.6524)
#
# An error report file with more information is saved as:
# /tmp/process-executionzWXy23/hs_err_pid6524.log
#
# Compiler replay data is saved as:
# [thread 6588 also had an error]
/tmp/process-executionzWXy23/replay_pid6524.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
```

Changing the internal JDK to the Temurin 11 distribution seems to remove the problem permanently:

```toml
[jvm]
tool_jdk = "temurin:1.11"
```

**Pants version**
2.11.0rc1

**OS**
MacOS

**Additional info**
Initially mentioned here: https://pantsbuild.slack.com/archives/C046T6T9U/p1649158967074989

--- a/src/python/pants/jvm/subsystems.py
+++ b/src/python/pants/jvm/subsystems.py
@@ -25,7 +25,7 @@ class JvmSubsystem(Subsystem):
 
     tool_jdk = StrOption(
         "--tool-jdk",
-        default="adopt:1.11",
+        default="temurin:1.11",
         help=softwrap(
             """
             The JDK to use when building and running Pants' internal JVM support code and other
@@ -36,7 +36,7 @@ class JvmSubsystem(Subsystem):
     )
     jdk = StrOption(
         "--jdk",
-        default="adopt:1.11",
+        default="temurin:1.11",
         help=softwrap(
             """
             The JDK to use.

